rootfs/etc/nginx/lua/balancer.lua

```diff
@@ -0,0 +1,112 @@
+local json = require "json"
+local b = require "ngx.balancer"
+
+local function compare(t, fn)
+    if #t == 0 then return nil, nil end
+    local key, value = 1, t[1]
+    for i = 2, #t do
+        if fn(value, t[i]) then
+            key, value = i, t[i]
+        end
+    end
+    return key, value
+end
+
+local http_host = ngx.var.http_host
+local remote_addr = ngx.var.remote_addr
+local proxy_upstream_name = ngx.var.proxy_upstream_name
+local shared_memory = ngx.shared.shared_memory;
+local cfgs_json = shared_memory:get("CFGS")
+if cfgs_json == nil then
+    ngx.status = 503
+    ngx.exit(ngx.status)
+end
+local cfgs = json.decode(cfgs_json)
```

Comment:
this might have significant performance penalty, because it is being done for every request

Reply:
True but I not found alternatives. Only shared dicts are visible worker wide.

Reply:
Here https://github.com/Shopify/ingress/pull/16/files#diff-b00d77a6df9c8c05a483044b08e6bc50R87 is how I did it. I'm curious what do you think about that.

Basically periodically in every worker read the config from shared dictionary, decode it and store it locally.

Then the balancer algorithm will use the parsed/decoded config from local cache.

Reply:
This is a very good implementation, congrats!

I understand proposed solution is not perfect but my intent was to add an experimental feature to be improved later.

Reply:
Thanks @valeriano-manassero! I made a PR at https://github.com/Shopify/shopify-docker-images/pull/612#issuecomment-370837614. If you agree with the ideas there and fine with the implementation I suggest we try to get it merged and continue adding the different LB algorithms you implemented here on top of it.

It already provides a mechanism to easily and cleanly add support for more LB algorithms but there's still room for improvement.

Reply:
PR link was incorrect in my comment above, it's not pointing to the right PR.